test/local/user_agent.js

```diff
@@ -559,6 +559,36 @@ describe('userAgent', () => {
   )
 
   it(
+    'recognises new Firefox-iOS user agents',
+    () => {
+      parserResult = {
+        ua: {
+          family: 'Other'
+        },
+        os: {
+          family: 'Other'
+        },
+        device: {
+          family: 'Other'
+        }
+      }
+      const context = {}
+      const userAgentString = 'Firefox-iOS-FxA/6.0b42 (iPhone 6S; iPhone OS 10.3) (Firefox)'
+      const result = userAgent.call(context, userAgentString, log)
+
+      assert.equal(result.uaBrowser, 'Firefox')
+      assert.equal(result.uaBrowserVersion, '6.0')
+      assert.equal(result.uaOS, 'iOS')
```

Comment:
This is a good point. I wasn't sure where those iPad/iPhone/iPod ones were coming from actually, nor the annoying Windows ones that contain the version. After a bit of a hunt it looks like they're from [`common_log_format.normalize_user_agent`](https://github.com/mozilla-services/lua_sandbox_extensions/blob/master/lpeg/modules/lpeg/common_log_format.lua#L423-L445) in `mozilla-services/lua_sandbox`, where they're defined in [`ua_os_matchers`](https://github.com/mozilla-services/lua_sandbox_extensions/blob/master/lpeg/modules/lpeg/common_log_format.lua#L359-L377):

```lua
local ua_os_matchers = {
    -- search, replace
      {"iPod"                    ,"iPod"         }
    , {"iPad"                    ,"iPad"         }
    , {"iPhone"                  ,"iPhone"       }
    , {"Android"                 ,"Android"      }
    ...
```

So we'll have to work round it in our metrics pipeline lua scripts unfortunately, I can open a PR. If we want to really fix it for all the data, it would require yet another re-import of the data and I'm really, really tired of those. 😩:smile: